File: angular/src/app/shared/services/teacher/models/create-update-teacher.dto.ts
import { EntityDto } from '@abp/ng.core';

export class CreateOrEditTeacherDto extends EntityDto<string>{

  name: string;
  birthday: Date;
  phoneNumber: string;
  email: string;
  address: string;
  degree: string;
  faculty: string;
  position: string;

  constructor(initial: Partial<CreateOrEditTeacherDto> = {}) {
    super(initial);
  }
}

File: aspnet-core/src/doan.ProjectManagement.Application.Contracts/Teachers/Dto/CreateUpdateTeacherDto.cs
using System;
using Volo.Abp.Application.Dtos;

namespace doan.ProjectManagement.Teachers.Dto
{
    public class CreateUpdateTeacherDto : EntityDto<Guid?>
    {
        public string Name { get; set; }
        public DateTime? Birthday { get; set; }
        public string PhoneNumber { get; set; }
        public string Email { get; set; }
        public string Address { get; set; }
        public string Degree { get; set; }
        public string Faculty { get; set; }
        public string Position { get; set; }
    }
}
